docs: Add a few security-related warnings to JWT auth docs (#3299)
apollographql/router: docs/source/configuration/authn-jwt.mdx

By enabling JWT authentication, you can block malicious traffic at the edge of your graph instead of relying on [header forwarding](./header-propagation/) to propagate tokens to your subgraphs.

> > ⚠️ Under all circumstances, it is important that subgraphs are only accessible via the router, and not directly. This is especially true if you are relying on JWT authentication in the router. Be sure to [secure your subgraphs](/federation/building-supergraphs/subgraphs-overview/#securing-your-subgraphs).

Reviewer: ```suggestion
> ⚠️ Under all circumstances, your subgraphs should be accessible _only_ via the router—not directly by clients. This is especially true if you're relying on JWT authentication in the router. Make sure to [secure your subgraphs](/federation/building-supergraphs/subgraphs-overview/#securing-your-subgraphs).
```

### Example: Forwarding claims to subgraphs

> > ⚠️ This example forwards JWT claims via headers. When combined with [header propagation](./header-propagation/) this can introduce a security risk. If you are relying on header-based forwarding, **be sure not to propagate matching headers**.

Reviewer: ```suggestion
> ⚠️ This example forwards JWT claims via headers. When combined with [header propagation](./header-propagation/) this can introduce a security risk. If you're relying on header-based forwarding, **make sure not to propagate matching headers**.
```

> > ⚠️ This example forwards JWT claims via headers. When combined with [header propagation](./header-propagation/) this can introduce a security risk. If you're relying on header-based forwarding, **make sure not to propagate matching headers**.

Reviewer 2: this requires an example, otherwise readers won't know what's the actual risk

Author: Example added—though I didn't go in-depth in mitigation. The idea was to add a quick-fix before #3224 can be handled, so I don't want to add a bunch of potential mitigations in the example (e.g., enforcing claim presence, deleting matching headers)
